app/services/metrics/development_cycle/per_product.rb

```diff
@@ -0,0 +1,42 @@
+module Metrics
+  module DevelopmentCycle
+    class PerProduct < Metrics::Base
+      private
+
+      def process
+        week_intervals.flat_map do |week|
+          interval = build_interval(week)
+          query(interval).map do |product, metric_value|
+            Metric.new(product, interval.first, metric_value)
+          end
+        end
+      end
+
+      def query(interval)
+        product = Product.find(@entity_id)
+
+        resolved_issues = product.jira_project.jira_issues.where(resolved_at: interval)
```

Comment:
I found that this method is a bit long and also `resolved_issues` is used too many times in and outside it.

If more code is needed in the future, before code smells, may be is good to consider for example doing a new **def** and use the `@resolved_issues `.

WDYT?